go/pkg/services/logcollector/server.go

```diff
@@ -418,48 +419,60 @@ func (s *Server) startLogStreaming(ctx context.Context,
 		return
 	}
 
-	// get logs from pod, and keep the stream open (follow)
-	podLogOptions := &v1.PodLogOptions{
-		Follow: true,
+	// open log file in read/write and append, to allow reading the logs while we write more logs to it
+	openFlags := os.O_RDWR | os.O_APPEND
+	file, err := os.OpenFile(logFilePath, openFlags, 0644)
+	if err != nil {
+		s.Logger.ErrorWithCtx(ctx, "Failed to open file", "err", err, "logFilePath", logFilePath)
+		return
 	}
-	restClientRequest := s.kubeClientSet.CoreV1().Pods(s.namespace).GetLogs(podName, podLogOptions)
+	defer file.Close() // nolint: errcheck
 
 	// initialize stream and error for the while loop
 	var (
-		streamErr error
-		stream    io.ReadCloser
+		stream         io.ReadCloser
+		streamErr      error
+		streamErrCount = 0
+		keepLogging    = true
 	)
 
-	// stream logs - retry if failed
-	err := common.RetryUntilSuccessful(15*time.Second, 3*time.Second, func() (bool, error) {
+	// get logs from pod, and keep the stream open (follow)
+	podLogOptions := &v1.PodLogOptions{
+		Follow: true,
+	}
+	restClientRequest := s.kubeClientSet.CoreV1().Pods(s.namespace).GetLogs(podName, podLogOptions)
+
+	// get the log stream - retry if failed
+	if err := common.RetryUntilSuccessful(15*time.Second, 3*time.Second, func() (bool, error) {
 		stream, streamErr = restClientRequest.Stream(ctx)
 		if streamErr != nil {
-			s.Logger.WarnWithCtx(ctx,
-				"Failed to get pod log stream, retrying",
-				"runUID", runUID,
-				"err", streamErr)
+
+			// first 3 errors are not logged - they are expected if pod is not ready yet
+			if streamErrCount > 3 {
+				s.Logger.WarnWithCtx(ctx,
+					"Failed to get pod log stream, retrying",
+					"runUID", runUID,
+					"err", streamErr)
+			}
```

Comment:
If you were preventing log spam - ignore me.
If not, maybe having the err log but debug (so less spammy) and expected would be nice
```suggestion
			if streamErrCount > 3 {
				s.Logger.WarnWithCtx(ctx,
					"Failed to get pod log stream, retrying",
					"runUID", runUID,
					"err", streamErr)
			} else {
				s.Logger.DebugWithCtx(ctx,
					"Failed to get pod log stream as pod is still starting up, retrying",
					"runUID", runUID,
					"err", streamErr)
			}
```

Reply:
Exactly prevented log spam, as I wrote above it:
```
first 3 errors are not logged - they are expected if pod is not ready yet
```